### PR TITLE
[dotnet] Add `[Uns|S]upportedOSPlatformAttribute` to manual bindings (part 1)

### PR DESCRIPTION
**Pull request by spouliot**
Once complete the `LegacyAttributes` can be enabled in introspection to
ensure no old attributes are introduced.

### CODE REVIEW

**Review by dalexsoto — APPROVED** [review]
Phew! This was a long one, amazing work! 👍

File: src/MLCompute/MLHelpers.cs [hunk 1]
```diff
@@ -124,8 +179,14 @@ public static string GetDebugDescription (this MLCSoftmaxOperation self)
 		}
 	}
 
-	[iOS (14,0)][TV (14,0)][Mac (10,16)]
+#if NET
+	[SupportedOSPlatform ("ios14.0")]
+	[SupportedOSPlatform ("tvos14.0")]
+	[SupportedOSPlatform ("macos11.0")]
```
Comment (stephen-hawley) [inline src/MLCompute/MLHelpers.cs:185]: Should this be 10.16?

File: src/MLCompute/MLHelpers.cs [hunk 2]
```diff
@@ -137,8 +198,14 @@ public static string GetDebugDescription (this MLCSampleMode self)
...
```
Comment (stephen-hawley) [inline src/MLCompute/MLHelpers.cs:204]: Same.

File: src/MLCompute/MLHelpers.cs [hunk 3]
```diff
@@ -150,8 +217,14 @@ public static string GetDebugDescription (this MLCLstmResultMode self)
 		}
 	}
 
-	[TV (14,5)][Mac (11,3)][iOS (14,5)]
+#if NET
+	[SupportedOSPlatform ("ios14.0")]
```
Comment (stephen-hawley) [inline src/MLCompute/MLHelpers.cs:221]: Why not 14.5, 11.3 and 14.5?

Comment (spouliot) [reply src/MLCompute/MLHelpers.cs]: that's a mistake that the test caught, fixed

File: src/MLCompute/MLHelpers.cs (hunk 1 above)
Comment (spouliot) [reply src/MLCompute/MLHelpers.cs:185]: no, 10.16 is 11.0 - only the first beta using 10.16
we kind of support both (in legacy) but there's no point in moving that forward (dotnet)

**Comment by spouliot** [comment]
unrelated, known issues
* monotouch-test/Mac [dotnet]/Debug [dotnet]: Failed (Test run crashed (exit code: 134). -> https://github.com/xamarin/maccore/issues/2413
* introspection/watchOS 32-bits - simulator/Debug (watchOS 5.0): LaunchFailure -> https://github.com/xamarin/maccore/issues/2411

**Pull request merged**